### PR TITLE
For #20259 - Add HistoryMetadataSuggestionProvider in AwesomeBar

### LINKED ISSUE #20259: Add HistoryMetadataSuggestionProvider in AwesomeBar

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #20259 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt [hunk 1]
```diff
@@ -117,6 +120,14 @@ class AwesomeBarView(
                 engineForSpeculativeConnects
             )
 
+        historyMetadataProvider =
+            HistoryMetadataSuggestionProvider(
+                historyStorage =components.core.historyStorage,
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt:125]: nit: space after `=`

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt [hunk 2]
```diff
@@ -117,6 +120,14 @@ class AwesomeBarView(
...
+                loadUrlUseCase = loadUrlUseCase,
+                icons = components.core.icons,
+                engine = engineForSpeculativeConnects
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt:128]: @Mugurell added a new parameter `maxNumberOfSuggestions` which we should set. The ticket (FNX-22340) wants us to display the top 3 results, so you could add `maxNumberOfSuggestions = 3`.

**Review by csadilek — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt (hunk 2 above)
Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt:128]: This is waiting for AC bump.

**Comment by grigoryk** [comment]
Together with @csadilek we traced the problem down to this one-liner: https://github.com/mozilla/application-services/pull/4330

**Comment by csadilek** [comment]
This will need https://github.com/mozilla-mobile/android-components/pull/10475 (which includes the A-S upgrade) to land.

**Comment by csadilek** [comment]
We still didn't catch the A-S fix and the latest A-C here is `92.0.20210721180104` but A-C upgrade only landed last night. So we need a `92.0.20210722+`.

**Pull request merged**